### DISCUSSION TITLE
What port is used to access the mailing server?

### DISCUSSION THREAD

**Discussion by osamadel** [ecederstrand/exchangelib#1191] [Q&A] [answered]
Hello

I was trying to send an email from Python using Microsoft Exchange and found this repo. Other methods used either SMTP or IMAP which use certain port numbers (i.e. 993). Now I'm using this library and it works on my machine but I need to run it on a secured machine with a Firewall and I'm not sure what the port number is to open in order to have access. I tried looking into the source code and couldn't quite get the method you use to communicate with the server.

Your help is much appreciated

**Comment by ecederstrand** [comment, marked as answer]
exchangelib uses the EWS API to communicate with the Exchange server. This API runs on top of the HTTPS protocol, which is normally served from port 443.

**Reply by osamadel** [reply]
Thank you
